runtime/interpreter/interpreter.go

```diff
@@ -3378,9 +3379,12 @@ var typeFunction = NewUnmeteredHostFunctionValue(
 
 		ty := typeParameterPair.Value
 
-		return TypeValue{
-			Type: ConvertSemaToStaticType(ty),
-		}
+		staticType := ConvertSemaToStaticType(ty)
+		return NewTypeValue(
+			invocation.Interpreter,
+			common.NewTypeMemoryUsage(staticType.String()),
+			func() StaticType { return staticType },
```

Comment:
At Bastian's suggestion, I'm going to leave the `TypeValue` metering in a half-done state because it's complex. Ticket for that: https://github.com/onflow/cadence/issues/1503.